### PR TITLE
don't override False with None in user upload

### PR DESCRIPTION
**Pull request by czue**
Somewhere in our excel parsing magic, certain values like "0" are automatically converted to `False`, but then `False` was getting converted back to `None` by this code, which then caused [this check](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/user_importer/validation.py#L187) to fail on valid "False" data.

##### FEATURE FLAG

Enable two-stage user provisioning (users confirm and set their own passwords via email).

##### PRODUCT DESCRIPTION

Now certain "falsey" values like "0" also work to set `is_account_confirmed` to false in bulk user upload, and not just "no".

### CODE REVIEW

**Comment by mkangia** [comment]
@czue build failure looks related
```
======================================================================
ERROR: corehq.apps.user_importer.tests.test_importer:TestUserBulkUpload.test_blank_is_active
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/user_importer/tests/test_importer.py", line 233, in test_blank_is_active
    [],
  File "/vendor/lib/python3.6/site-packages/celery/local.py", line 193, in __call__
    return self._get_current_object()(*a, **kw)
  File "/vendor/lib/python3.6/site-packages/celery/app/task.py", line 379, in __call__
    return self.run(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/user_importer/tasks.py", line 28, in import_users_and_groups
    update_progress=functools.partial(_update_progress, start=len(group_specs))
  File "/mnt/commcare-hq-ro/corehq/apps/user_importer/importer.py", line 368, in create_or_update_users_and_groups
    user.is_active = is_active
  File "jsonobject/base.pyx", line 327, in jsonobject.base.JsonObjectBase.__setattr__
  File "jsonobject/base_properties.pyx", line 79, in jsonobject.base_properties.JsonProperty.__set__
  File "jsonobject/base.pyx", line 299, in jsonobject.base.JsonObjectBase.__setitem__
  File "jsonobject/base.pyx", line 281, in jsonobject.base.JsonObjectBase.__unwrap
  File "jsonobject/base_properties.pyx", line 283, in jsonobject.base_properties.AssertTypeProperty.unwrap
  File "jsonobject/base_properties.pyx", line 269, in jsonobject.base_properties.AssertTypeProperty.assert_type
jsonobject.exceptions.BadValueError: '' not of type <class 'bool'>
----------------------------------------------------------------------
```

**Comment by czue** [comment]
thanks, https://github.com/dimagi/commcare-hq/pull/27211/commits/32c9289aa6504ba557a114fe7c7949fe94df88a3

File: corehq/apps/user_importer/helpers.py [hunk 1]
```diff
@@ -5,4 +5,7 @@ def spec_value_to_boolean_or_none(user_spec_dict, key):
     value = user_spec_dict.get(key, None)
     if value and isinstance(value, str):
         return string_to_boolean(value)
-    return value
+    elif isinstance(value, bool):
+        return value
+    else:
+        return None
```
Comment (mkangia) [inline corehq/apps/user_importer/helpers.py:11]: the last else block looks redundant.

**Review by mkangia — APPROVED** [review]

File: corehq/apps/user_importer/helpers.py (hunk 1 above)
Comment (czue) [reply corehq/apps/user_importer/helpers.py:11]: Good point, though in this case I think it's actually maybe better to be explicit. Do you disagree?

Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:11]: Being explicit is great and i am always in favour of it, just that here it's in the end of the method and would be clear without the else as well.
There is also an option to entirely skip `return None` which would make this implicit, which i am not asking for here.

Comment (mkangia) [reply corehq/apps/user_importer/helpers.py:11]: (fine to punt)

**Pull request merged**